api/grpcserver/grpc.go

```diff
@@ -44,35 +46,27 @@ func (s *Server) Start() <-chan struct{} {
 			return nil
 		})),
 	)
-
-	started := make(chan struct{})
-	go s.startInternal(started)
-
-	return started
-}
-
-// Blocking, should be called in a goroutine.
-func (s *Server) startInternal(started chan<- struct{}) {
 	lis, err := net.Listen("tcp", s.Listener)
 	if err != nil {
 		s.logger.Error("error listening: %v", err)
-		return
+		return err
 	}
 	reflection.Register(s.GrpcServer)
-	s.logger.Info("starting new grpc server on %s", s.Listener)
-	close(started)
-	if err := s.GrpcServer.Serve(lis); err != nil {
-		s.logger.Error("error stopping grpc server: %v", err)
-	}
+	s.grp.Go(func() error {
+		if err := s.GrpcServer.Serve(lis); err != nil {
+			s.logger.Error("error stopping grpc server: %v", err)
+			return err
+		}
+		return nil
+	})
+	return nil
 }
 
 // Close stops the server.
 func (s *Server) Close() error {
 	s.logger.Info("stopping the grpc server")
 	s.GrpcServer.Stop()
```

Comment:
Consider a graceful stop to allow pending GRPC calls to finish:
```suggestion
	s.GrpcServer.GracefulStop()
```

Reply:
I'm don't think just calling GracefulStop would work well with our streaming APIs, I think if you had a stream open the server would never shutdown. The approach to handling this, according to this [ticket](https://github.com/grpc/grpc-go/issues/2888), is to first call GracefulStop, wait some time and then call Stop. I can add a ticket for it, it seems to be out of scope for this PR.

Reply:
Thanks, didn't know that! Do you think it's worth creating an issue to implement the proposed graceful shutdown later?

Reply:
Sure will do!

Reply:
https://github.com/spacemeshos/go-spacemesh/issues/4867